neow3j-test/src/main/java/io/neow3j/test/ContractTestExtension.java

```diff
@@ -0,0 +1,213 @@
+package io.neow3j.test;
+
+import io.neow3j.compiler.CompilationUnit;
+import io.neow3j.compiler.Compiler;
+import io.neow3j.contract.ContractUtils;
+import io.neow3j.contract.SmartContract;
+import io.neow3j.crypto.WIF;
+import io.neow3j.protocol.Neow3jExpress;
+import io.neow3j.protocol.ObjectMapperFactory;
+import io.neow3j.protocol.core.response.NeoApplicationLog;
+import io.neow3j.protocol.http.HttpService;
+import io.neow3j.script.VerificationScript;
+import io.neow3j.types.Hash160;
+import io.neow3j.types.Hash256;
+import io.neow3j.utils.Await;
+import io.neow3j.utils.Numeric;
+import io.neow3j.wallet.Account;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static io.neow3j.test.NeoExpressTestContainer.CONTAINER_WORKDIR;
+
+public class ContractTestExtension implements BeforeAllCallback, AfterAllCallback {
+
+    private static final String DEFAULT_NEOXP_CONFIG = "default.neo-express";
+
+    // Extension Context Store
+    final static String CONTAINER_STORE_KEY = "neoExpressContainer";
+    final static String NEOW3J_STORE_KEY = "neow3j";
+    final static String CONTRACT_STORE_KEY = "contractHash";
+
+    private String neoxpConfigFileName = DEFAULT_NEOXP_CONFIG;
+
+    // Properties
+//    private final static String PROP_FILE_NAME = "test.properties";
+//    private final static String DEFAULT_PROP_FILE_NAME = "test-default.properties";
+//    private final static String GENESIS_PROPS_PREFIX = "genesis";
+//    private final static String ACC_PROPS_PREFIX = "account";
+//    private final static String GENESIS_ADDR = GENESIS_PROPS_PREFIX + ".address";
+//    private final static String GENESIS_SCRIPT_HASH = GENESIS_PROPS_PREFIX + ".script-hash";
+//    private final static String ACC_1_ADDR = ACC_PROPS_PREFIX + "1.address";
+//    private final static String ACC_1_SCRIPT_HASH = ACC_PROPS_PREFIX + "1.script-hash";
+//    private final static String ACC_1_PRIV_KEY = ACC_PROPS_PREFIX + "1.private-key";
+
+    //    private Properties properties;
+    private NeoExpressTestContainer container;
+    private Neow3jExpress neow3j;
+    private SmartContract contractUnderTest;
+
+//    public ContractTestExtension() {
```

Comment:
oh.... now I get what you were trying to do here.... so, basically, you wanted to "read" files from resources where a certain prefix/suffix is set, and then inject those to the neo-express. Is that right?

In addition to that, I think the test framework should rely on an existing neo-express config file, if specified by the user... but maybe this is something up to discussion in the future.

Reply:
First, I wanted to use a properties file where we would set the default account information (e.g., private keys) that are included in the default neo-express config. Later, I changed to the approach of actually reading the account information from the config JSON. Then, this properties information became obsolete. I left it here in case we might make use of properties in the future. Though, I can remove it now since it is available in at least one commit.

Reply:
The user can define her own neoxp config that will overwrite the default one.